### ISSUE TITLE
Crash when saving an empty Graph 2D

### ISSUE THREAD

**Issue by guiguem** [project8/katydid#75]
After the two cuts on points during the track processing, if there is no more points, Katydid crashes when attempting to create a TGraph2D (see attachment).
[KTcrash.txt](https://github.com/project8/katydid/files/522429/KTcrash.txt)

**Comment by nsoblath** [comment]
@guiguem: can you please also attach the configuration file you were using?

**Comment by guiguem** [comment]
The data file is rid000001097-2016.09.13.22.55.32.411.MAT
[AlgorithmOptimization copy.txt](https://github.com/project8/katydid/files/522480/AlgorithmOptimization.copy.txt)

**Comment by evzayas** [comment]
This should probably be handled in the writer, since a sparse waterfall plot with 0 points is a bogus track but still an important possible output of the track processing analysis. In [this commit](https://github.com/project8/katydid/commit/4b716128f14285ce47d7fbc8cebcdbfae6c8be70) from last week on the Event Viewer branch, I simply had it throw a warning instead of trying to create the empty TGraph. Does this seem like a reasonable fix?

**Comment by guiguem** [comment]
For what my opinion worths, that sounds reasonable... I added this piece of code in my branch, and fixed my problem...

**Comment by nsoblath** [comment]
Which piece of code?  Can you push that change to GitHub?  If it's fixing a bug that's relevant outside of your branch I will probably cherry pick it into a hotfix.

**Comment by evzayas** [comment]
Mathieu is talking about adding these lines in KTROOTTreeTypeWriterEventAnalysis.cc around 300, before the TGraph2D is initialized:

```
if( points.size() == 0 )
{
    KTWARN(publog, "The Sparse Waterfall Plot has no points! Will not write it");
    return;
}
```

**Comment by nsoblath** [comment]
This has been fixed in the update I just made (commit 2ae94b1).  It has been merged into develop and included in the release version 2.6.3.
